### DISCUSSION TITLE
Takes a long time to block after adding to the blocklist

### DISCUSSION THREAD

**Discussion by imflash217** [SelfControlApp/selfcontrol#751] [Q&A]
Hi,
After the url is added to the blocklist, it takes a very very long time (more than an hour) to block the website. 
**Is this because I have added 100s of website to the blocklist?**
**Is there a way to make the blocking start instantly?**
Thanks

**Comment by cstigler** [comment]
Hi @imflash217 , wow that is a long time! Yes, it probably is due in part to the size of your blocklist, unfortunately, although I've tested blocklists up to 1000 sites long and never had them take longer than a minute or so on my computer.

Are you up-to-date running version 4.0.3? Also, are you running a blocklist block (i.e. blocking specific websites), or an allowlist block (i.e only allowing specific websites)? If it's an allowlist block, disabling the "Allowlist blocks should include linked sites" option in the settings would help.

**Reply by imflash217** [reply]
> Are you up-to-date running version 4.0.3?

No. I am currently running `v3.0.3` (and I cannot upgrade to `v4.0.3` as my time ris already running and upgrade will not allow me to add new sites)

> although I've tested blocklists up to 1000 sites long and never had them take longer than a minute or so on my computer.

I guess the reason is because I have manually added 1000s of sites one-by-one and now if I add a new site I have to close the app and reopen it again to add another one.....and the block happens after a long time. I see speedup (& less frequent closing-reopening of app) if I leave the app closed for some time and then add many sites one by one ....but again the same symptoms arise after few additions. Seems like an issue with my Mac (maybe) based on your comments. Will check it agian with newer `v4.0.3`.

**Reply by cstigler** [reply]
Ah, yeah that's your issue. 3.x is much, much, much slower than 4.x at adding sites to your blocklist. 4.x will probably be much better.
